### ISSUE TITLE
Option for adding custom payment gateways

### ISSUE THREAD

**Issue by ssensalo** [davidgewirtz/seamless-donations#7]
If a developer needs to add a custom payment gateway besides Paypal or Stripe, this would allow them to integrate their own payment option maybe with a local bank or another payment gateway provider.

**Comment by davidgewirtz** [comment]
Sadly, doing that is far harder than you might think. I'm still struggling
to get basic PayPal working reliably. When I adopted the plugin, PayPal was
baked into every nook and cranny, so extracting and generalizing it is
almost a complete rewrite. I did add Stripe, which taught me a lot about
how non-general the problem is. I am aware of a library called (I think)
Omnipay, and I'm thinking about integrating it, which would allow for other
gateways. But even it isn't a simple API. Net-net, since this is far from
my pays-the-bills gig and this is a barfingly big job, it's way on the back
burner.

*David Gewirtz*
Educator, Advisor, Columnist, Geek
Author of *The Flexible Enterprise* and *How To Save Jobs*
Read my columns at ZDNet DIY-IT <http://www.zdnet.com/blog/diy-it/> and ZDNet
Government <http://www.zdnet.com/blog/government/>
Read my lab notes at ZATZLabs <http://zatzlabs.com/lab-notes/> and my videos
on YouTube <https://www.youtube.com/user/DavidGewirtzTV/>
Mobile: +1 321-722-4620

Twitter <http://twitter.com/davidgewirtz> | Facebook
<https://www.facebook.com/davidgewirtz> | Instagram
<https://www.instagram.com/davidgewirtz/> | LinkedIn
<https://www.linkedin.com/in/davidgewirtz> | YouTube
<http://youtube.com/davidgewirtztv>

**Comment by ssensalo** [comment]
Yes, I have watched your youtube videos they are quite exhaustive I was able to get a clear understanding around the plugin, great job @davidgewirtz. 
Let me look more into the omnipay API next week.
True this issue can't necessarily be generalized, but like you see for woocommerce its possible to add a custom payment gateway option and I think this plugin is really great we can enable more fine-grained control of the payment gateway for developers. I think it would also increase usage to an extent.

**Comment by davidgewirtz** [comment]
Feel free to dig around. Don't work off the Github version. Git has now a
few times eaten my dev version, so I've given up on fighting with it.
Instead, the almost current version is on the WordPress repo. Not too many
changes since then outside of the paypal PHP files, so if you decide on a
tweak you want me to incorporate, I can do so pretty easily.

*David Gewirtz*
Educator, Advisor, Columnist, Geek
Author of *The Flexible Enterprise* and *How To Save Jobs*
Read my columns at ZDNet DIY-IT <http://www.zdnet.com/blog/diy-it/> and ZDNet
Government <http://www.zdnet.com/blog/government/>
Read my lab notes at ZATZLabs <http://zatzlabs.com/lab-notes/> and my videos
on YouTube <https://www.youtube.com/user/DavidGewirtzTV/>
Mobile: +1 321-722-4620

Twitter <http://twitter.com/davidgewirtz> | Facebook
<https://www.facebook.com/davidgewirtz> | Instagram
<https://www.instagram.com/davidgewirtz/> | LinkedIn
<https://www.linkedin.com/in/davidgewirtz> | YouTube
<http://youtube.com/davidgewirtztv>
